rclpy_examples/listener_py.py

```diff
@@ -27,16 +27,28 @@ def chatter_callback(msg):
 def main(args=None):
     if args is None:
         args = sys.argv
+    else:
+        from rclpy.impl.rmw_implementation_tools import select_rmw_implementation
+        select_rmw_implementation(args[0])
 
-    rclpy.init(args)
+    rclpy.init()
 
     node = rclpy.create_node('listener')
 
-    sub = node.create_subscription(String, 'chatter', chatter_callback, qos_profile_default)
+    sub = node.create_subscription(
+        String, 'chatter', chatter_callback, qos_profile_default)
     assert sub  # prevent unused warning
 
     while rclpy.ok():
         rclpy.spin_once(node)
 
+
+class main_for_rmw_impl_class(object):
+    def __getattr__(self, key):
+        return main([key])
```

Comment:
`sys.argv` which the `args` keyword argument defaults to has a different semantic. It contains the script name as the first entry. Therefore I think this needs to be refactored somehow to have the same semantic when being invoked from the command line as well as the entry point.